### PR TITLE
feat: add support for Edge Functions

### PR DESCRIPTION
**Pull request by ascorbic**
### Summary
Adds beta support for Netlify Edge Functions. This includes support for deploying middleware to the edge, but also running the full site using edge runtime. To enable this during the beta, set `NEXT_USE_NETLIFY_EDGE` in the UI or cli.

### Test plan

1. Visit the deploy previews for the middleware and server-components demos (coming soon)

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

loads of capybaras in the sun

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

File: plugin/src/index.ts [hunk 1]
```diff
@@ -100,6 +97,15 @@ const plugin: NetlifyPlugin = {
       nextConfig: { basePath, i18n, trailingSlash, appDir },
       buildId,
     })
+
+    if (process.env.NEXT_USE_NETLIFY_EDGE) {
+      console.log(outdent`
+        ✨ Deploying to ${greenBright`Netlify Edge Functions`} ✨
+        This feature is in beta. Please share your feedback here: https://ntl.fyi/next-netlify-edge
```
Comment (orinokai) [inline plugin/src/index.ts:104]: This link seems to just point to the Netlify homepage

File: plugin/src/templates/edge/ipx.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import type { Context } from 'netlify:edge'
+
+/**
+ * Implement content negotiation for images
+ */
+
+const handler = async (req: Request, context: Context) => {
+  const { searchParams } = new URL(req.url)
+  const accept = new Accepts(req.headers)
+  const type = accept.types(['avif', 'webp'])
```
Comment (orinokai) [inline plugin/src/templates/edge/ipx.ts:11]: Just AVIF and WEBP?

**Review by orinokai — DISMISSED** [review]
Looks awesome to me! Just that minor docs issue and image format question.

File: plugin/src/index.ts (hunk 1 above)
Comment (ascorbic) [reply plugin/src/index.ts:104]: We'll create this when we launch. There's a bullet in the launch checklist

File: plugin/src/templates/edge/ipx.ts (hunk 2 above)
Comment (ascorbic) [reply plugin/src/templates/edge/ipx.ts:11]: Yeah. If they don't support either of those then we don't emforce a type, and let ipx serve the original format

File: plugin/src/templates/edge/ipx.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return new Response('Invalid request', {
+      status: 400,
+    })
+  }
+
+  const modifiers = [`w_${width}`, `q_${quality}`]
+
+  if (type) {
+    modifiers.push(`f_${type}`)
+  }
```
Comment (ascorbic) [inline plugin/src/templates/edge/ipx.ts:27]: ...here is where we add the modifier if they support a modern format

Comment (orinokai) [reply plugin/src/templates/edge/ipx.ts:27]: Got it, thanks 👍🏼

**Review by orinokai — APPROVED** [review]
🚢

**Pull request merged**